File: example/WebContent/js/controllers/songs_controller.js
$(function() {
   var rootContext = '/songs';
   var controller = new Uno.Controller(rootContext);
   var viewManager = new Uno.ViewManager(rootContext);

   controller.bind('/index', function(data) {
      var viewData = {
         title: 'Search songs',
         songs: Song.findAll(),
         filter: {}
      };

      viewManager.loadTemplate('index', viewData);
   });

   controller.bind('/show', function(data) {
      var song = null;

      if (data.params) {
         var viewData = {
            title: 'Show song',
            song: Song.findById(data.params.id)
         };

         viewManager.loadTemplate('show', viewData);
      } else {
         controller.throwError(400);
      }
   });

   controller.bind('/delete', function(data) {
      if (data.params) {
         var song = Song.findById(data.params.id);
         song.remove();
         controller.redirect('/index');
      } else {
         controller.throwError(400);
      }
   });

   controller.bind('/new', function(data) {
      var viewData = {
         title: 'New song',
         song: (data.params && data.params.song) ? data.params.song : {}
      };

      viewManager.loadTemplate('maitenance', viewData);
   });

   controller.bind('/edit', function(data) {
      var song;

      if (data.params && data.params.id) {
         song = Song.findById(data.params.id);
      } else if (data.song) {
         song = data.params.song;
      }

      if (song) {
         var viewData = {
            title: 'Edit song',
            song: song
         };

         viewManager.loadTemplate('maitenance', viewData);
      } else {
         controller.throwError(400);
      }
   });

   controller.bind('/save', function(data) {
      if (data.params) {
         song = new Song(data.params);

         if (song.save()) {
            controller.redirect('/index');
         } else {
            var previousState = (song.id) ? '/edit' : '/new';
            controller.forward(previousState, {
               song: song
            });
         }
      } else {
         controller.throwError(400);
      }
   });

   controller.bind('/search', function(data) {
      if (data.params) {
         var filter = {
            id: (data.params.id) ? data.params.id : undefined,
            title: (data.params.title) ? data.params.title : undefined
         };

         var viewData = {
            title: 'Search songs',
            songs: Song.findWithFilter(filter),
            filter: filter
         };

         viewManager.loadTemplate('index', viewData);
      } else {
         controller.throwError(400);
      }
   });
});

File: example/src/main/java/unosample/controller/SongsController.java
package unosample.controller;

import java.io.IOException;
import java.util.List;

import javax.servlet.http.HttpServletResponse;

import org.json.JSONException;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import unosample.controller.filters.SongSearchFilter;
import unosample.model.entity.Song;
import unosample.model.service.SongService;

@Controller
@RequestMapping("/songs")
public class SongsController {
    @Autowired
    private SongService songService;
    
    @RequestMapping("/index")
    public void findAll(HttpServletResponse resp) throws JSONException, IOException {
        List<Song> songs = songService.findAll();
        JSONObject jsonObject = new JSONObject();
        
        jsonObject.put("songs", songs);
        
        resp.getWriter().print(jsonObject);
    }
    
    @RequestMapping("/show")
    public void findById(@RequestParam Long id, HttpServletResponse resp) throws IOException {
        Song song = songService.findById(id);
        
        if (song == null) {
            resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
        } else {
            JSONObject jsonObject = new JSONObject(song);
            resp.getWriter().print(jsonObject);
        }
    }
    
    @RequestMapping("/delete")
    public void delete(@RequestParam String entityData, HttpServletResponse resp) throws IOException {
        try {
            JSONObject jsonObject = new JSONObject(entityData);
            Long id = jsonObject.getLong("id");
            Song song = songService.findById(id);
            
            if (song == null) {
                resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
            } else {
                songService.deleteWithId(id);
            }
        } catch (JSONException e) {
            resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
        }
    }
    
    @RequestMapping("/save")
    public void save(@RequestParam String entityData, HttpServletResponse resp) throws IOException {
        try {
            Song song = new Song(entityData);
            songService.saveOrUpdate(song);
            
            JSONObject jsonObject = new JSONObject();
            jsonObject.put("id", song.getId());
            
            resp.getWriter().print(jsonObject);
        } catch (JSONException e) {
            resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);            
        }
    }
    
    @RequestMapping("/search")
    public void search(@RequestParam String filter, HttpServletResponse resp) throws JSONException, IOException {
        List<Song> songs = null;
        JSONObject jsonObject = null;
        
        try {
            SongSearchFilter searchFilter = new SongSearchFilter(filter);
            songs = songService.findWithFilter(searchFilter);
            jsonObject = new JSONObject();
        } catch (JSONException e) {
            resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
            return;
        }
        
        jsonObject.put("songs", songs);
        resp.getWriter().print(jsonObject);
    }
}
